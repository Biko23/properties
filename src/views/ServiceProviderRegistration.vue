<template>
  <div>
    <v-container>
      <v-row>
        <v-col cols="12" sm="12" md="12" xl="12">
          <h3 style="text-align: center; margin-top: 20px">
            Your profile service provider
          </h3>
          <v-form id="form" style="background-color: #e7f0ff">
            <v-container>
              <v-row>
                <v-col cols="12" sm="12" md="12">
                  <p>Full name</p>
                  <v-text-field
                    label="Full name"
                    placeholder="Enter your full name"
                    solo
                    v-model="newVendor.vendor_name"
                  ></v-text-field>
                </v-col>
              </v-row>
              <!-- <v-row>
                <v-col cols="12" sm="12" md="12">
                  <p>Username:</p>
                  <v-text-field
                    label="Username"
                    placeholder="Username"
                    solo
                   
                  ></v-text-field>
                </v-col>
              </v-row> -->
              <v-row>
                <v-col cols="12" sm="12" md="12">
                  <p>Telephone:</p>
                  <v-text-field
                    label="Telephone"
                    placeholder="Telephone number"
                    solo
                    v-model="newVendor.vendor_primary_phone_number"
                    readonly
                  ></v-text-field>
                </v-col>
              </v-row>
              <v-row>
                <v-col cols="12" sm="12" md="12">
                  <v-select
                    :items="allVendorsCategories"
                    label="Select Category"
                    solo
                    v-model="newVendor.category_type"
                  ></v-select>
                </v-col>
              </v-row>
              <v-row>
                <v-col cols="12" sm="12" md="12">
                  <p>Email</p>
                  <v-text-field
                    label="Email "
                    placeholder="forexample@gmail.com"
                    solo
                    v-model="newVendor.vendor_primary_email"
                    readonly
                  ></v-text-field>
                </v-col>
              </v-row>
              <v-row>
                <v-col cols="12" sm="12" md="12">
                  <v-btn color="primary" block @click="postingVendor"
                    >Register As Provider</v-btn
                  >
                </v-col>
              </v-row>
            </v-container>
          </v-form>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script>
import { mapActions, mapGetters } from "vuex";
import BottonNav from "../components/BottonNav.vue";
import MainNav from "../components/MainNav.vue";
import TopNav from "../components/TopNav.vue";
export default {
  components: { TopNav, MainNav, BottonNav },
  name: "VendorProfile",

  data: () => ({
    vendorCategories: [],
    editedIndex: -1,
    newVendor: {
      vendor_name: "",
      vendor_primary_phone_number: "",
      vendor_secondary_phone_number: "",
      vendor_primary_email: "",
      vendor_secondary_email: "",
      category_type: "",
    },
    defaultItem: {
      vendor_name: "",
      vendor_primary_phone_number: "",
      vendor_secondary_phone_number: "",
      vendor_primary_email: "",
      vendor_secondary_email: "",
      category_type: "",
    },
    valid: true,
    vendorRules: {
      required: (v) => !!v || "Required field",
      min: (v) => (v && v.length > 3) || "Value should be greater than 4",
      phoneLength: (v) =>
        (v && v.length >= 10) || "Number should not be less than 9",
      email: (v) => /.+@.+\..+/.test(v) || "E-mail must be valid",
    },
  }),
  computed: {
    ...mapGetters(["allVendorsCategories", "currentLoggedinUser"]),
  },

  created() {
    this.fetchVendorsCategories();
  },
  mounted() {
    this.loadVendorData();
  },
  methods: {
    ...mapActions(["fetchVendorsCategories", "postVendor", "postAUserLog"]),
    async postingVendor() {
       this.postAUserLog({
            activity: `Registered as a Service provider Page`,
            button_clicked: "Register As A Provider Button"
        });
      try {
        const response = await this.postVendor(this.newVendor);
        if (response.status === 201 || response.status === 200) {
          this.$router.push("/register");
        }
      } catch (error) {
        console.log(error);
      }
    },
    loadVendorData() {
      this.newVendor = Object.assign({}, this.currentLoggedinUser);
    },
  },
};
</script>

<style scoped>
#form {
  background-color: #e7f0ff;
  margin-left: auto;
  margin-right: auto;
  width: 900px;
  border-radius: 6px;
}
#reach {
  text-align: center;
  font-size: 0.9rem;
  margin-top: 30px;
}
</style>