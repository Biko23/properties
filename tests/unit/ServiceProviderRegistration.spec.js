import { createLocalVue, mount } from '@vue/test-utils'
import ServiceProviderRegistration from '@/views/ServiceProviderRegistration.vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueRouter from 'vue-router'


describe('ServiceProviderRegistration.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)
    let actions
    let getters
    let store
    let vuetify
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
        actions = {
            fetchVendorsCategories: jest.fn(), 
            postVendor: jest.fn(), 
            postAUserLog: jest.fn()
        }
        getters = {
            allVendorsCategories: () => ({}), 
            currentLoggedinUser: () => ({})
        }
        store = new Vuex.Store({
            actions,
            getters
        })
    })

    const mountFactory = options => {
        return mount(ServiceProviderRegistration, {
            localVue,
            vuetify,
            store,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFactory({})
        expect(wrapper.html()).toMatchSnapshot()
    })
})

