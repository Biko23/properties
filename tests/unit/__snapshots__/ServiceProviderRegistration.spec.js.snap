// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceProviderRegistration.vue should have a match snapshot 1`] = `
<div>
  <base-dialog message="" title=""></base-dialog>
  <div class="container">
    <div class="row">
      <div class="col-sm-12 col-md-12 col-xl-12 col-12">
        <h3 style="text-align: center; margin-top: 20px;">
          Your profile service provider
        </h3>
        <form novalidate="novalidate" id="form" class="v-form" style="background-color: rgb(231, 240, 255);">
          <div class="container">
            <div class="row">
              <div class="col-sm-12 col-md-12 col-12">
                <p>Full name</p>
                <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <div class="v-text-field__slot"><label for="input-5" class="v-label theme--light" style="left: 0px; position: absolute;">Full name</label><input id="input-5" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="row">
              <div class="col-sm-12 col-md-12 col-12">
                <p>Telephone:</p>
                <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <div class="v-text-field__slot"><label for="input-8" class="v-label theme--light" style="left: 0px; position: absolute;">Telephone</label><input id="input-8" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="row">
              <div class="col-sm-12 col-md-12 col-12">
                <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select">
                  <div class="v-input__control">
                    <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-11" class="v-input__slot">
                      <div class="v-select__slot"><label for="input-11" class="v-label theme--light" style="left: 0px; position: absolute;">Select Category</label>
                        <div class="v-select__selections"><input id="input-11" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                        </div><input type="hidden">
                      </div>
                      <div class="v-menu">
                        <!---->
                      </div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="row">
              <div class="col-sm-12 col-md-12 col-12">
                <p>Email</p>
                <div class="v-input v-input--is-readonly theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <div class="v-text-field__slot"><label for="input-16" class="v-label theme--light" style="left: 0px; position: absolute;">Email </label><input id="input-16" readonly="readonly" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="row">
              <div class="col-sm-12 col-md-12 col-12"><button type="button" class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">Register As Provider</span></button></div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
